python/hopsworks_common/connection.py

```diff
@@ -410,7 +424,9 @@ def connect(self) -> None:
             self._variable_api = variable_api.VariableApi()
             usage.init_usage(self._host, self._variable_api.get_version("hopsworks"))
 
-            if self._project:
+            if self._project and self._variable_api.get_variable(
```

Comment:
Why is it not enough to just fix it with this change? Are the other changes related to this fix?

Reply:
No, they are not related. I can extract them to a separate PR.

Reply:
Please pull [these changes](https://github.com/logicalclocks/hopsworks-api/commit/6db402f668e923d3365d8b8a499f254e33ff9092) before merging the rest of the PR since the code here is not correct.